src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/CaseAssignedRoleService.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.finrem.caseorchestration.client.DataStoreClient;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseAssignedUserRolesResource;
+
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.finrem.caseorchestration.OrchestrationConstants.YES_VALUE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT_REPRESENTED;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APP_SOLICITOR_POLICY;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CASE_ROLE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CONSENTED_RESPONDENT_REPRESENTED;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CONTESTED_RESPONDENT_REPRESENTED;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESP_SOLICITOR_POLICY;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class CaseAssignedRoleService {
+
+    private final DataStoreClient dataStoreClient;
+    private final CaseDataService caseDataService;
+    private final AuthTokenGenerator authTokenGenerator;
+
+    public Map<String, Object> setCaseAssignedUserRole(CaseDetails caseDetails,
+                                                       String authToken) {
+
+        CaseAssignedUserRolesResource resource = dataStoreClient.getCaseAssignedUserRoles(caseDetails.getId(),
```

Comment:
Related to the change for the data store client, change this to `List.of(caseDetails.getId().toString())`

Reply:
Modified the method signature, and updated this line of code as well.